src/core/reactor.ts

```diff
@@ -1293,6 +1310,31 @@ protected _getFirstReactionOrMutation(): Reaction<any> | undefined {
         // this._sourcePort.delete(src);
     }
 
+    private _uncheckedDisconnect<R extends Present, S extends R>(src: IOPort<S>, dst:IOPort<R>) {
+        Log.debug(this, () => "disconnecting " + src + " and " + dst);
+        this._dependencyGraph.removeEdge(dst, src);
+        // Register receiver for value propagation.
+        let writer = dst.asWritable(this._getKey(dst));
+        src.getManager(this._getKey(src)).delReceiver
+            (writer as WritablePort<S>);
+        this._dependencyGraph.removeNode(src);
+        this._dependencyGraph.removeNode(dst);
+        
+    }
+
+    public canDisconnect<R extends Present, S extends R>
+        (src: IOPort<S>, dst: IOPort<R>) {
+        if (src === dst) {
```

Comment:
I don't think we need to handle this case separately.

Reply:
I merged the condition for ```canDisconnect``` into the ```_disconnect``` function to simplify it and allow for ```_uncheckedDisconnect``` to be executed.